### PR TITLE
WIP-mobile-robot-unicycle

### PR DESCRIPTION
**Pull request by johnaoga**
This is a new benchmark inspired by [this work of Antoine Girard](https://www.sciencedirect.com/science/article/pii/S2405896322012150)

Preview https://dionysos-dev.github.io/Dionysos.jl/previews/PR354

### CODE REVIEW

File: docs/src/examples/solvers/Path planning sgmpc.jl [hunk 1]
```diff
@@ -57,7 +57,7 @@ concrete_system = concrete_problem.system;
 
 # Definition of the grid of the state-space on which the abstraction is based (origin `x0` and state-space discretization `h`):
 x0 = SVector(0.0, 0.0, 0.0);
-h = SVector(0.2, 0.2, 0.2);
+h = SVector(0.1, 0.1, 0.2);
```
Comment (adrienbanse) [inline docs/src/examples/solvers/Path planning sgmpc.jl:60]: @johnaoga cc my e-mail, just revert if you want to go back to the coarser abstraction

Comment (johnaoga) [reply docs/src/examples/solvers/Path planning sgmpc.jl:60]: Thank you! No worries!

**Comment by blegat** [comment]
The failure should be fixed by https://github.com/JuliaIntervals/IntervalLinearAlgebra.jl/pull/135

**Comment by blegat** [comment]
You can rebase on master to fix the documentation build so that we can see the preview

**Comment by blegat** [comment]
We can see now that the controller says that it succeeded but there is not trajectory displayed at the end:
https://dionysos-dev.github.io/Dionysos.jl/previews/PR354/generated/Path%20planning%20sgmpc/
Anyone has an idea what's wrong ?

**Comment by adrienbanse** [comment]
@blegat @johnaoga  The trajectory is only one point

`Dionysos.System.Control_trajectory{SVector{3, Float64}, Any}(Dionysos.System.Trajectory{SVector{3, Float64}}(SVector{3, Float64}[[0.5, 0.5, 0.0]]), Dionysos.System.Trajectory{Any}(Any[]))`

Screenshot 2024-06-19 at 09 48 28

**Comment by johnaoga** [comment]
Thank you @adrienbanse for the investigation.

But, That's exactly what surprises me, since the starting point is set to be (0.5, -1.5), I would have expected to have at least 2 points in the trajectory, the starting point and the point (0.5, 0.5).

@adrienbanse @blegat But from your perspective do you think its answer is okay or it maybe a bug from the algorithm? In which case I could explore that direction

**Comment by adrienbanse** [comment]
@johnaoga The initial state is in the target set so the controller directly stops

**Comment by adrienbanse** [comment]
@blegat @JulienCalbert Any idea why the examples throws `compute_controller_reach! terminated without covering init set` ?

**Comment by adrienbanse** [comment]
The example probably fails because of the self-loops in the automation. To mitigate this, let's try
- A larger timestep
- A finer abstraction

**Comment by adrienbanse** [comment]
@johnaoga The example is fixed, let's try to clean as much as possible the code and the presentation before merging

Screenshot 2024-06-28 at 11 47 57

**Comment by johnaoga** [comment]
thank you @adrienbanse, i will clean (if still necessary) and add comments

**Comment by blegat** [comment]
@johnaoga is this closed by https://github.com/dionysos-dev/Dionysos.jl/pull/383 ?

**Comment by johnaoga** [comment]
> @johnaoga is this closed by #383 ?

I'm closing it now. 

But the problem I'm trying to solve here is a bit different from the one in the master. Here, I have a stage cost, a terminal cost (i.e, an explicit definition of the time horizon) and a terminal constraint sets. I'll open a new PR whenever these capacities are available in Dionysos.

**Pull request closed without merging**